lib/rpc/SingleJsonRpcProvider.ts

```diff
@@ -472,14 +465,41 @@ export class SingleJsonRpcProvider extends StaticJsonRpcProvider {
     )
   }
 
-  override call(
+  override call(transaction: Deferrable<TransactionRequest>, blockTag?: BlockTag | Promise<BlockTag>): Promise<string> {
+    return this.wrappedFunctionCall(CallType.NORMAL, 'call', super.call.bind(this), transaction, blockTag)
+  }
+
+  override send(method: string, params: Array<any>): Promise<any> {
+    return this.wrappedFunctionCall(CallType.NORMAL, 'send', super.send.bind(this), method, params)
+  }
+
+  ///////////////////// Begin of special functions /////////////////////
+
+  // Wrap another layer only for the sake of ease unit testing.
```

Comment:
i dont see any unit tests for this one so not sure why we created this functions? Also why is it easier to unit test i am a little confused here, we are adding boilerplate code is my undersatnding right?

Reply:
It's certainly not adding boilerplate code. The reason we have a `_getBlockNumber()` besides `getBlockNumber()` is that we want to mock the output (such as provider network call response) without removing the call of `wrappedFunctionCall()`, which contains important logic that a lot of UT cases will check against. Yes the only purpose is to make the mocking easier for UT.

I don't want to do this for other APIs since it just pollute the code too much. So I only did it for `getBlockerNumber()` API. It's good enough to represent major use cases though.